### ISSUE TITLE
Clashes due to multiple Set-Cookie headers 

### ISSUE THREAD

**Issue by MarkTsengTW** [adafruit/Adafruit_CircuitPython_Requests#104]
Hi, first time submitting an issue on Github so I hope this makes sense!

The function _parse_headers appears to handle multiple headers with the same name by successively overwriting them, leaving only the final header with that name.

I'm trying to read the headers from a requests.post() response where the response included two 'Set-Cookie' headers. 

The headers are:
```
Set-Cookie: JSESSIONID=...
Set-Cookie: SERVERID=...
```

This works on standard Python requests when accessing response.headers, and I've verified using Wireshark that the correct headers are being sent to the CircuitPython board too.

However, on CircuitPython, response.headers only yields:

`Set-Cookie: SERVERID=...
`

I'm assuming that because _headers is a dictionary and duplicate keys aren't allowed, _parse_headers doesn't support multiple cookies. I wish I could suggest a solution to this but as a beginner it's beyond me!

I'm not sure if this issue affects requests.session. I came across it working with the Adafruit Matrix Portal, where requests.session isn't supported.

**Comment by MarkTsengTW** [comment]
After tea and meditation I have arrived at an answer. 

The problematic area is:

```
            if title and content:
                # enforce that all headers are lowercase
                title = str(title, "utf-8").lower()
                content = str(content, "utf-8")
                if title == "content-length":
                    self._remaining = int(content)
                if title == "transfer-encoding":
                    self._chunked = content.strip().lower() == "chunked"
                self._headers[title] = content

```

The following works for my project:

```
            if title and content:
                # enforce that all headers are lowercase
                title = str(title, "utf-8").lower()
                content = str(content, "utf-8")
                if title == "content-length":
                    self._remaining = int(content)
                if title == "transfer-encoding":
                    self._chunked = content.strip().lower() == "chunked"
                if title == "set-cookie":
                    if "set-cookie" in self._headers:
                        self._headers["set-cookie"] = self._headers["set-cookie"] + ", " + content
                    else:
                        self._headers[title] = content
```

It checks for an existing set-cookie header and, if one exists, appends the next cookie to the previous one, separated by a comma. 

I do not know if this breaks other projects or other parts of the library. Another issue is that it does not behave the same way as requests, preserving the headers as originally sent (which the user would reasonably expect). Finally, I am not sure if the code is the most efficient or correct way to get the result.

**Comment by askpatrickw** [comment]
@MarkTsengTW you seem to be doing a great job narrowing it down, do you have a suggested fix?

**Comment by askpatrickw** [comment]
Best way to get feedback is to submit a PR so we can clearly see the diffs, make suggestions and give it a test run.

**Comment by tekktrik** [comment]
I was looking into this a little bit ago, and yeah, I agree comma separated is probably a good solution (and I think what CPython does as well).  The people that typically work on the core may have more information or thoughts, but since this library gets frozen into the firmware for a few boards, than my money is that a change to add cookies as a comma-separated string is going to use less memory than trying to implement something like the `cookie` property or `CookieJar` that CPython uses.

I agree with @askpatrickw - a PR for people to review is going to be the best path forward!

**Comment by MarkTsengTW** [comment]
Thanks for the encouragement @askpatrickw and @tekktrik. I've made a pull request [here](https://github.com/MarkTsengTW/Adafruit_CircuitPython_Requests/pull/1) with an improved version of my code.

**Comment by tekktrik** [comment]
Hey @MarkTsengTW, I followed up on your PR about submitting it the this repository, let us know if you have any questions!

**Comment by FoamyGuy** [comment]
resolved by #108
